File: src/test/java/de/lukaseichler/recurrentneuralnetwork/structure/SmoothingLayerTest.java
package de.lukaseichler.recurrentneuralnetwork.structure;

import org.testng.annotations.Test;
import de.lukaseichler.recurrentneuralnetwork.activation.TanhActivation;

import static org.assertj.core.api.Assertions.assertThat;

public class SmoothingLayerTest {

    @Test
    public void creatingWithoutParameter() throws Exception {
        assertThat(new SmoothingLayer().getNeuronCount()).isEqualTo(0);
    }

    @Test
    public void creatingWithParameter() throws Exception {
        SmoothingLayer smoothingLayer = new SmoothingLayer(2);
        assertThat(smoothingLayer.getNeuronCount()).isEqualTo(2);
    }

    @Test
    public void activationFunction() throws Exception {
        SmoothingLayer smoothingLayer = new SmoothingLayer(2);
        assertThat(smoothingLayer.getActivationFunction()).hasSameClassAs(new TanhActivation());
    }
}

File: src/test/scala/de/lukaseichler/recurrentneuralnetwork/structure/SmoothingLayerTest.scala
package de.lukaseichler.recurrentneuralnetwork.structure

import de.lukaseichler.recurrentneuralnetwork.activation.TanhActivation
import org.scalatest.Matchers
import org.scalatest.testng.TestNGSuite
import org.testng.annotations.Test

/**
 * @author leichler
 */
class SmoothingLayerTest extends TestNGSuite with Matchers {

  @Test
  def creatingWithoutParameters(): Unit = {
    new SmoothingLayer().getNeuronCount should be(0)
  }

  @Test
  def creatingWithParameter() {
    val smoothingLayer: SmoothingLayer = new SmoothingLayer(2)
    smoothingLayer.getNeuronCount should be(2)
  }

  @Test
  def activationFunction(): Unit = {
    val smoothingLayer: SmoothingLayer = new SmoothingLayer(2)
    smoothingLayer.getActivationFunction shouldBe a[TanhActivation]
  }

}
